lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1907,10 +1936,22 @@ def _resolve_depenency_map(
 
         for req in dep_exc.criterion.iter_requirement():
             error_msg_lines.append(
-                f'* {req.fqcn!s}:{req.ver!s}'
+                f'* {req !s}'
             )
         error_msg_lines.append(pre_release_hint)
 
         raise AnsibleError('\n'.join(error_msg_lines)) from dep_exc
+    except CollectionDependencyResolverRuntimeError as resolvelib_crash:
+        raise AnsibleError(
+            f'Collection dependency resolution crashed: {resolvelib_crash !s}.'
+            '\n\n'
+            "If you are seeing this, something's gone terribly wrong causing "
+            'the `resolvelib` library to misbehave. Please make sure that '
+            'the `resolvelib` version you have in the same environment that '
+            'runs `ansible-galaxy` is supported. Then, restart the command '
+            'with `ANSIBLE_DEBUG=1` environment variable and `-vvvvv`. Once '
+            'you have all the logs, please, file an issue with a reproducer '
+            'per <https://sscce.org>.',
```

Comment:
Recommending a specific level of verbosity seems useful, but the other specifics in the error message seem much less helpful. We guard against unsupported resolvelib versions, ANSIBLE_DEBUG won't emit more useful information currently, and I think it would make more sense to put general advice like referencing https://sscce.org in an the issue templates, not individual error messages.

```suggestion
            f"Collection dependency resolution crashed: {resolvelib_crash !s}."
            "\n\n"
            "Please re-run the command with `-vvvvv` and report this as a bug. "
```

Reply:
Okay, agreed. I might've been overthinking this.